### PR TITLE
Adding API server support for service account

### PR DESCRIPTION
**Pull request by blublinsky**
## Why are these changes needed?

Current implementation of the operator supports usage of the service accounts, while API server has no ability to specify it. This PR adds Service account parameter to head and worker nodes definitions and propagate them to corresponding pod templates

## Checks

- [ x] I've made sure the tests are passing. 
- Testing Strategy
   - [ x] Unit tests
   - [ ] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by blublinsky** [comment]
@kevin85421, can you, please, take a look at the errors? They all come from the operator, not my code

**Comment by kevin85421** [comment]
#1145 is merged. Would you mind rebasing with the master branch? However, I found the nightly compatibility to be quite unstable. There may have been some recent breaking changes in the Ray master branch.

File: apiserver/pkg/model/converter.go [hunk 1]
```diff
@@ -214,7 +218,12 @@ func PopulateWorkerNodeSpec(specs []v1alpha1.WorkerGroupSpec) []*api.WorkerGroup
 			}
 			workerNodeSpec.Environment = env
 		}
-		// Resources.
+
+		// If
```
Comment (Jeffwan) [inline apiserver/pkg/model/converter.go:222]: Please remove or finish the comment?

File: proto/cluster.proto [hunk 2]
```diff
@@ -236,12 +239,15 @@ message WorkerGroupSpec {
   map<string, string> ray_start_params = 7;
   // Optional. The volumes mount to worker pods
   repeated Volume volumes = 8;
+  // Optional. ServiceAccount used by head pod
+  // Note that the service account has to be created prior to usage here
+  string serviceAccount = 9;
```
Comment (Jeffwan) [inline proto/cluster.proto:244]: let's follow the naming conversions and use `service_account` here.

**Review by Jeffwan — CHANGES_REQUESTED** [review]

File: proto/cluster.proto (hunk 2 above)
Comment (blublinsky) [reply proto/cluster.proto:244]: Done

File: apiserver/pkg/model/converter.go (hunk 1 above)
Comment (blublinsky) [reply apiserver/pkg/model/converter.go:222]: fixed

**Pull request closed without merging**